tests/smoke/test_virtwho_smoke.py

```diff
@@ -0,0 +1,223 @@
+
+    def test_vw_rhsm_options(self):
+        self.system_unregister(self.ssh_host())
+        register_type = self.register_config['type']
+        self.vw_option_add("rhsm_hostname", self.register_config['server'], self.config_file)
+        self.vw_option_add("rhsm_port", "443", self.config_file)
+        self.vw_option_add("rhsm_prefix", self.register_config['prefix'], self.config_file)
+        self.vw_option_add("rhsm_username", self.register_config['username'], self.config_file)
+        self.vw_option_add("rhsm_password", self.register_config['password'], self.config_file)
+        data, tty_output, rhsm_output = self.vw_start(exp_send=1)
+        assert(self.op_normal_value(data, exp_error=0, exp_thread=1, exp_send=1))
+        self.system_register(self.ssh_host(), register_type, self.register_config)
+
+    def test_vw_http_proxy(self):
+        squid_server = "{0}:{1}".format(deploy.proxy.server, deploy.proxy.port)
+        proxy_type = ['http_proxy', 'https_proxy']
+        for type in proxy_type:
+            logger.info(">>> run virt-who to check {0}".format(type))
+            if type == "http_proxy":
+                proxy_url = "http://{0}".format(squid_server)
+            else:
+                proxy_url = "https://{0}".format(squid_server)
+            self.vw_option_add(type, proxy_url, "/etc/sysconfig/virt-who")
+            data, tty_output, rhsm_output = self.vw_start(exp_send=1)
+            assert (self.op_normal_value(data, exp_error=0, exp_thread=1, exp_send=1))
+            assert (self.vw_msg_search(rhsm_output, "Using proxy.*{0}".format(squid_server)))
+
+    def test_vw_hypervisor_id(self):
+        register_owner = self.register_config['owner']
+        hypervisor_type = self.get_config('hypervisor_type')
+        if hypervisor_type in ('esx', 'rhevm'):
+            hypervisor_ids = ['uuid', 'hostname', 'hwuuid']
+        else:
+            hypervisor_ids = ['uuid', 'hostname']
+        for hypervisor_id in hypervisor_ids:
+            if 'hwuuid' in hypervisor_ids:
+                if hypervisor_id == "uuid":
+                    ids = [self.host_uuid, self.host_name, self.host_hwuuid]
+                elif hypervisor_id == "hostname":
+                    ids = [self.host_name, self.host_uuid, self.host_hwuuid]
+                else:
+                    ids = [self.host_hwuuid, self.host_uuid, self.host_name]
+            else:
+                if hypervisor_id == "uuid":
+                    ids = [self.host_uuid, self.host_name]
+                else:
+                    ids = [self.host_name, self.host_uuid]
+            logger.info(">>> check hypervisor_id in rhsm.log")
+            self.vw_option_add("hypervisor_id", hypervisor_id, self.config_file)
+            data, tty_output, rhsm_output = self.vw_start()
+            assert (self.op_normal_value(data, exp_error=0, exp_thread=1, exp_send=1))
+            assert (ids[0] in data[register_owner].keys())
```

Comment:
Could we have some solution to optimize these code, it looks so long and hard understand:)